src/macros.rs

```diff
@@ -79,25 +79,24 @@ macro_rules! redis_module {
             _argv: *mut *mut raw::RedisModuleString,
             _argc: c_int,
         ) -> c_int {
-            {
-                // We use an explicit block here to make sure all memory allocated before we
-                // switch to the Redis allocator will be out of scope and thus deallocated.
-                let module_name = CString::new($module_name).unwrap();
-                let module_version = $module_version as c_int;
-
-                if unsafe { raw::Export_RedisModule_Init(
-                    ctx,
-                    module_name.as_ptr(),
-                    module_version,
-                    raw::REDISMODULE_APIVER_1 as c_int,
-                ) } == raw::Status::Err as c_int { return raw::Status::Err as c_int; }
+            let mut name_buffer = [0; 64];
+            let mut dest = name_buffer.as_mut_ptr();
+            for byte in $module_name.chars() {
+                unsafe {
+                    *dest = byte as i8;
+                    dest = dest.add(1);
+                }
             }
```

Comment:
I'd prefer to do this as follows (there is no need to distinguish between UTF-8 and plain bytes here, so no need to call `chars()`:
```
        std::ptr::copy(
            module_name.as_ptr(),
            name_buffer.as_mut_ptr(),
            module_name.len(),
        );
```